app/code/Magento/Catalog/Model/ResourceModel/AbstractResource.php

```diff
@@ -466,9 +466,9 @@ protected function _getOrigObject($object)
      * Checks also attribute's store scope:
      * We should insert on duplicate key update values if we unchecked 'STORE VIEW' checkbox in store view.
      *
-     * @param AbstractAttribute $attribute
-     * @param mixed $value New value of the attribute.
-     * @param array &$origData
+     * @param  AbstractAttribute $attribute
```

Comment:
I think the static tests might be failing because of the extra space here and the following lines though I am not too sure.

```
 469 | ERROR   | [x] Method arguments visual alignment must be
     |         |     consistent
```

Reply:
I was getting the same error before the spaces were added, so I added the extra spaces so that they aligned with the return too. Didn't make a difference though.

What's odd is that if I run that static test locally, I don't get the error :( 

(I'll revert the commit anyway since it didn't make a difference)

Reply:
He @dmanners , thanks for looking into that! 

I took another look and thanks to comments on this other PR  (https://github.com/magento/magento2/pull/23304) it led me to find that it was the "&$origdata" causing that error online 469, phpcs must've been reporting the wrong line number (oddly!). I've made some further commits and that error is now cleared along with the other warnings, however one part is still failing: 

```
----------------------------------------------------------------------
 18 | WARNING | Abstract classes MUST NOT be marked as public @api.
----------------------------------------------------------------------
```

I feel like this error is slightly "above my paygrade" so to speak and it might be worth someone Magento side making a ruling on it. The obvious solution is to remove "api" from the class comment, but that's also obviously not a decision to take lightly given how these classes are referred to in the devdocs https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/ (though perhaps I'm also overthinking it :P)

Not sure how we proceed with this one!

Let me know what you think

Reply:
Hey @dmanners 

I've been able to fix the static test by adding: 

phpcs:disable Magento2.Classes.AbstractApi

To the top of Magento\Catalog\Model\ResourceModel\AbstractResource, like in some of the other classes in core Magento (e.g. Magento\Ui\Component\Form\Element\AbstractOptionsField)

This means all tests are passing now :)